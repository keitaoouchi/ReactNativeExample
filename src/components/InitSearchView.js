import React from "react";
import {Text, TouchableWithoutFeedback, View} from "react-native";
import Workaround from "../misc/Workaround";
import Color from "../misc/Color";
import ListenOnSpotifyButton from "./ListenOnSpotifyButton";
import dismissKeyboard from "dismissKeyboard";

export default InitSearchView = () => {
  return (
    <TouchableWithoutFeedback onPress={dismissKeyboard}>
      <View style={style.root}>
        <Text style={style.mainText}>検索してみよう。</Text>
        <Text style={style.subText}>好きなアーティストを探してみて</Text>
        <ListenOnSpotifyButton
          url="https://open.spotify.com/"
          title="Spotifyを開く"
          color={Color.pink}
        />
      </View>
    </TouchableWithoutFeedback>
  );
};

const style = {
  root: {
    flex: 1,
    flexDirection: 'column',
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: Color.black,
    ...Workaround.root,
  },
  mainText: {
    color: 'white',
    fontSize: 32,
    fontWeight: 'bold',
    margin: 16,
  },
  subText: {
    color: 'white',
    opacity: 0.7,
    fontSize: 14,
    marginLeft: 16,
    marginRight: 16,
    marginBottom: 32,
  },
};