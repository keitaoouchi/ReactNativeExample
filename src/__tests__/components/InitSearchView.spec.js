import "react-native";
import React from "react";
import renderer from "react-test-renderer";
import InitSearchView from "../../../src/components/InitSearchView";

describe('InitSearchView', () => {

  it('renders correctly', () => {
    const tree = renderer.create(
      <InitSearchView />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});