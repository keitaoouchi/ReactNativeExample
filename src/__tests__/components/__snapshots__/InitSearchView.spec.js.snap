// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InitSearchView renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hitSlop={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#181818",
      "flex": 1,
      "flexDirection": "column",
      "justifyContent": "center",
    }
  }
  testID={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "white",
        "fontSize": 32,
        "fontWeight": "bold",
        "margin": 16,
      }
    }
  >
    検索してみよう。
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "white",
        "fontSize": 14,
        "marginBottom": 32,
        "marginLeft": 16,
        "marginRight": 16,
        "opacity": 0.7,
      }
    }
  >
    好きなアーティストを探してみて
  </Text>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    isTVSelectable={true}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#1ed760",
            "borderRadius": 22,
            "height": 44,
            "margin": 12,
            "padding": 8,
            "width": 200,
          },
          Object {
            "backgroundColor": "#a82c9b",
          },
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "white",
            "fontSize": 16,
            "fontWeight": "bold",
            "margin": 4,
            "textAlign": "center",
          }
        }
      >
        Spotifyを開く
      </Text>
    </View>
  </View>
</View>
`;
